### DISCUSSION TITLE
Including smoke parameters in boundary conditions

### DISCUSSION THREAD

**Discussion by DavidHuber-NOAA** [ufs-community/ufs-srweather-app#897] [Q&A]
The RAP wrfprs files contain a field, MASSDEN, which is surface smoke concentration.  How do I extract this data from these files with chgres so it can be used as a boundary condition for the SRW?  I'm hoping I could just edit a few scripts, like [UFS_UTILS/parm/varmap_tables/GFSphys_var_map.txt](UFS_UTILS/blob/develop/parm/varmap_tables/GFSphys_var_map.txt), but I'm guessing I will need to add some code to [UFS_UTILS/sorc/chgres_cube/atmosphere.f90](https://github.com/ufs-community/UFS_UTILS/blob/develop/sorc/chgres_cube.fd/atmosphere.F90) to extract this field.  Any guidance would be much appreciated for myself and @JohanaRomeroAlvarez.

**Comment by natalie-perlin** [comment]
@DavidHuber-NOAA - 
Just responding here that your request has been received by the AUS Team. We will have to determine a right person in the community to answer this question about using chgres to extract MASSDEN variable

**Reply by DavidHuber-NOAA** [reply]
@natalie-perlin Thanks for the FYI.  I did have some luck doing this.  Though the work is not complete, I have a good understanding now of how to do this and started the work in this branch: [feature/add_massden](https://github.com/DavidHuber-NOAA/UFS_UTILS/tree/feature/add_massden).  As-is, this branch is able to read in the parameter, but the output is all fill.

To summarize the work I did so far, I added massden/smoke to the list of parameters in parm/varmap_tables/GFSphys_var_map.txt, added the GRIB2 octets for MASSDEN to atm_input_data.F90, and converted mass density (kg/m^3) to the RRFS-expected mixing ration (micrograms/kg).

Looking at the code changes, I think I may just need to change the missing pressure levels method in GFSphys_var_map.txt from `set_to_fill` to `intrp`.  I will give that a try and see if that returns a valid result.

**Reply by gspetro-NOAA** [reply]
Hi @DavidHuber-NOAA ! 

Just wanted to check in and see whether you succeeded in resolving your question or need further assistance. If you figured it out, I can close the discussion as resolved. 

Best,
Gillian

**Reply by DavidHuber-NOAA** [reply]
Hi @gspetro-NOAA!  I was not able to fully resolve this, but I am no longer working on it, so I am OK with closing it.

Just to add a final update, I was able to get an output, but it was coming out ~5 orders of magnitude smaller than expected.
